pkg/ottl/ottlfuncs/func_scale.go

```diff
@@ -0,0 +1,134 @@
+package ottlfuncs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+type ScaleArguments[K any] struct {
+	Value      ottl.Getter[K]
+	Multiplier float64
+}
+
+func NewScaleFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("Scale", &ScaleArguments[K]{}, createScaleFunction[K])
+}
+
+func createScaleFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*ScaleArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("ScaleFactory args must be of type *ScaleArguments[K]")
+	}
+
+	return Scale(args.Value, args.Multiplier)
+}
+
+func Scale[K any](value ottl.Getter[K], multiplier float64) (ottl.ExprFunc[K], error) {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		get, err := value.Get(ctx, tCtx)
+		if err != nil {
+			return nil, err
+		}
+		if floatVal, ok := get.(float64); ok {
```

Comment:
I think we should go for a type switch here, e.g.:

```golang
switch val := get.(type) {
    case float64:
        return val * multiplier, nil
    [...]
}
```

Reply:
thanks for the suggestion - adapted